### PR TITLE
Changing some verbiage and adding logic to add additional messaging b…

### PR DESCRIPTION
**Pull request by kylesoskin**
…ased on which identifiers a user may be missing

## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary

This change adds support to show an "it's ok" message to the error message shown to a user on the 526 flow if they are missing required identifiers. Previously there was concern that user did not know that they were not required to know the values of the things we were telling them were missing. Adding this message should alleviate stress or extra effort on the users part prior to calling to resolve issues. 

This change also changes to state they are available 24/7. 

I am on the Benefits and Claims team, 526 team 2 (carbs), and we own this flow/code and its maintenance. 

No flipper used.

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
[department-of-veterans-affairs/va.gov-team#89576](https://github.com/department-of-veterans-affairs/va.gov-team/issues/89576)

## Testing done

- _Describe what the old behavior was prior to the change_
- _Describe the steps required to verify your changes are working as expected_
- _Describe the tests completed and the results_
- _Exclusively stating 'Specs and automated tests passing' is NOT acceptable as appropriate testing
- _Optionally, provide a link to your [test plan](https://depo-platform-documentation.scrollhelp.site/developer-docs/create-a-test-plan-in-testrail) and [test execution records](https://depo-platform-documentation.scrollhelp.site/developer-docs/execute-tests-in-testrail)_

## Screenshots
Screenshot 2024-10-02 at 9 12 36 AM
Screenshot 2024-10-02 at 9 10 41 AM
Screenshot 2024-10-02 at 9 09 38 AM

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  | image|Screenshot 2024-10-02 at 9 38 47 AM |
| Desktop | image| Screenshot 2024-10-02 at 9 10 41 AM|

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/disability-benefits/all-claims/containers/Missing526Identifiers.jsx [hunk 1]
```diff
@@ -58,6 +74,7 @@ const displayContent = (title, form526RequiredIdentifers) => {
   const missingIdentifiers = filterMissingIdentifiers(
     form526RequiredIdentifers,
   );
+  const itsOkDisplayMessage = okMessageToDisplayText(missingIdentifiers);
```
Comment (freeheeling) [inline src/applications/disability-benefits/all-claims/containers/Missing526Identifiers.jsx:77]: Happy to see 'Identifiers' spelled correctly on L77, given the typo on L75.

**Review by freeheeling — DISMISSED** [review]
LGTM

File: src/applications/disability-benefits/all-claims/containers/Missing526Identifiers.jsx (hunk 1 above)
Comment (kylesoskin) [reply src/applications/disability-benefits/all-claims/containers/Missing526Identifiers.jsx:77]: oh shoot. I didn't touch.change that but.... we probably should

**Review by pjhill — DISMISSED** [review]

File: src/applications/disability-benefits/all-claims/tests/containers/Missing526Identifiers.unit.spec.jsx [hunk 2]
```diff
@@ -98,4 +98,112 @@ describe('Form 526 Missing Identifiers Error Message', () => {
...
+        edipi: true,
+      },
+    };
+
+    it('returns the missing birlsId OK message', () => {
+      const tree = render(<Missing526Identifiers {...props} />);
+      const messageParagraph = tree.container.querySelector('p');
+
+      expect(messageParagraph.textContent).match(
+        /It's ok if you don't know your BIRLS ID./,
```
Comment (freeheeling) [inline src/applications/disability-benefits/all-claims/tests/containers/Missing526Identifiers.unit.spec.jsx:119]: All instances of 'ok' should be uppercased to match casing in `okMessageToDisplayText`.

File: src/applications/disability-benefits/all-claims/containers/Missing526Identifiers.jsx (hunk 1 above)
Comment (freeheeling) [reply src/applications/disability-benefits/all-claims/containers/Missing526Identifiers.jsx:77]: Your `form526RequiredIdentifiers` updates in [104a741](https://github.com/department-of-veterans-affairs/vets-website/pull/32162/commits/104a7417668aae4a471b0519f7aed3183b254ee1) only included those in the spec. Did you want to change the typos on L73 and L75 in this file as well?

**Review by freeheeling — APPROVED** [review]
Looks good, with the understanding that the screenshots were captured prior to the commit that upcased 'ok'.

File: src/applications/disability-benefits/all-claims/tests/containers/Missing526Identifiers.unit.spec.jsx [hunk 3]
```diff
@@ -98,4 +98,112 @@ describe('Form 526 Missing Identifiers Error Message', () => {
...
+      form526RequiredIdentifiers: {
+        participantId: true,
+        birlsId: false,
+        ssn: true,
+        birthDate: true,
+        edipi: false,
+      },
+    };
+
+    it('returns the missing edipi and birls OK message', () => {
```
Comment (freeheeling) [inline src/applications/disability-benefits/all-claims/tests/containers/Missing526Identifiers.unit.spec.jsx:158]: birlsId

File: src/applications/disability-benefits/all-claims/containers/Missing526Identifiers.jsx [hunk 4]
```diff
@@ -16,28 +24,36 @@ const titleLowerCase = (title = '') => {
...
-  };
-
   const readableIdentifiers = missingIdentifiers.map(
     idName => READABLE_IDENTIFIER_MAPPING[idName],
   );
 
   return readableList(readableIdentifiers);
 };
 
+const okMessageToDisplayText = missingIdentifiers => {
```
Comment (NB28VT) [inline src/applications/disability-benefits/all-claims/containers/Missing526Identifiers.jsx:41]: Nit non-blocking, might consider renaming this is a little confusing at first

File: src/applications/disability-benefits/all-claims/containers/Missing526Identifiers.jsx [hunk 5]
```diff
@@ -16,28 +24,36 @@ const titleLowerCase = (title = '') => {
...
+  if (!missingIdentifiers) return '';
+
+  const OK_IF_NOT_KNOWN = ['participantId', 'birlsId', 'edipi'];
+
+  const readableIdentifiers = missingIdentifiers
+    .filter(id => OK_IF_NOT_KNOWN.includes(id))
+    .map(idName => READABLE_IDENTIFIER_MAPPING[idName]);
+  if (readableIdentifiers.length === 0) return '';
+
+  return ` It's OK if you don't know your ${readableList(
```
Comment (NB28VT) [inline src/applications/disability-benefits/all-claims/containers/Missing526Identifiers.jsx:51]: Is there a reason for having the space at the beginning of this string?

File: src/applications/disability-benefits/all-claims/containers/Missing526Identifiers.jsx [hunk 6]
```diff
@@ -54,10 +70,11 @@ Alert.propTypes = {
   title: PropTypes.string,
 };
 
-const displayContent = (title, form526RequiredIdentifers) => {
+const displayContent = (title, form526RequiredIdentifiers) => {
```
Comment (NB28VT) [inline src/applications/disability-benefits/all-claims/containers/Missing526Identifiers.jsx:73]: Lol I'm guessing this was my typo 🤦

File: src/applications/disability-benefits/all-claims/containers/Missing526Identifiers.jsx (hunk 5 above)
Comment (NB28VT) [reply src/applications/disability-benefits/all-claims/containers/Missing526Identifiers.jsx:51]: Oh I see now, yeah this makes sense

**Review by NB28VT — APPROVED** [review]
Nice, this all makes sense to me based on my limited recollection of the change I did; just one nitpick which is def not blocking.

The tests look good actually, a lot better than some of the tests I've seen in this repo nice and clean

File: src/applications/disability-benefits/all-claims/tests/containers/Missing526Identifiers.unit.spec.jsx [hunk 7]
```diff
@@ -98,4 +98,112 @@ describe('Form 526 Missing Identifiers Error Message', () => {
...
+    it('returns the missing edipi, birlsId, participant ID OK message', () => {
+      const tree = render(<Missing526Identifiers {...props} />);
+      const messageParagraph = tree.container.querySelector('p');
+
+      expect(messageParagraph.textContent).match(
+        /It's OK if you don't know your Participant ID, BIRLS ID, or EDIPI./,
+      );
+    });
+  });
+
```
[the remark this answers is not in the document]
Comment (kylesoskin) [reply src/applications/disability-benefits/all-claims/tests/containers/Missing526Identifiers.unit.spec.jsx:189]: I was just testing a few combos, but yea if there is only that one remaining to exhaust all possible combos I will add it in another PR and link here then resolve this comment

File: src/applications/disability-benefits/all-claims/tests/containers/Missing526Identifiers.unit.spec.jsx (hunk 3 above)
Comment (kylesoskin) [reply src/applications/disability-benefits/all-claims/tests/containers/Missing526Identifiers.unit.spec.jsx:158]: Will fix and link in upcoming PR

File: src/applications/disability-benefits/all-claims/containers/Missing526Identifiers.jsx (hunk 4 above)
Comment (kylesoskin) [reply src/applications/disability-benefits/all-claims/containers/Missing526Identifiers.jsx:41]: Will consider renaming and link in upcoming PR

File: src/applications/disability-benefits/all-claims/containers/Missing526Identifiers.jsx (hunk 1 above)
Comment (kylesoskin) [reply src/applications/disability-benefits/all-claims/containers/Missing526Identifiers.jsx]: I did a global replace and change before merging so this should be fixed now

File: src/applications/disability-benefits/all-claims/containers/Missing526Identifiers.jsx (hunk 4 above)
Comment (kylesoskin) [reply src/applications/disability-benefits/all-claims/containers/Missing526Identifiers.jsx:41]: https://github.com/department-of-veterans-affairs/vets-website/pull/32569

File: src/applications/disability-benefits/all-claims/tests/containers/Missing526Identifiers.unit.spec.jsx (hunk 3 above)
Comment (kylesoskin) [reply src/applications/disability-benefits/all-claims/tests/containers/Missing526Identifiers.unit.spec.jsx:158]: https://github.com/department-of-veterans-affairs/vets-website/pull/32569

File: src/applications/disability-benefits/all-claims/tests/containers/Missing526Identifiers.unit.spec.jsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (kylesoskin) [reply src/applications/disability-benefits/all-claims/tests/containers/Missing526Identifiers.unit.spec.jsx:189]: https://github.com/department-of-veterans-affairs/vets-website/pull/32569

**Pull request merged**